src/app/modules/shared/components/breadcrumb/breadcrumb.service.ts

```diff
@@ -0,0 +1,54 @@
+import { Injectable, EventEmitter } from '@angular/core'
+import { Router, ActivatedRouteSnapshot, Event, NavigationEnd, UrlSegment } from '@angular/router'
+import { Breadcrumb } from './breadcrumb.model'
+
+@Injectable()
+export class BreadcrumbService {
+  breadcrumbChanged = new EventEmitter<Breadcrumb[]>(false);
+
+  private breadcrumbs: Breadcrumb[] = [];
+
+  constructor(private router: Router) {
+    this.router.events.subscribe((routeEvent) => { this.onRouteEvent(routeEvent) });
+  }
+    
+  onRouteEvent(routeEvent: Event) {
```

Comment:
The naming `onRouteEvent` also comes from declarative programming. In rxjs, we are not dealing with events, but instead reacting to observable streams